fix issue #38032

fnst-yule opened the pull request:
to allow specifying num_samples to RandomSampler when repl…

lw committed:
Update TensorPipe submodule (#38923)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/38923

Test Plan: We'll see on CircleCI

Reviewed By: beauby

Differential Revision: D21703670

fbshipit-source-id: fd477486226303130906d669b0e9a1c888cfeee0

ShawnZhong committed:
Rename `torch._C.Generator` to `torch.Generator` (#38773)

Summary:
Fix https://github.com/pytorch/pytorch/issues/26528
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38773

Differential Revision: D21701053

Pulled By: pbelevich

fbshipit-source-id: 57632ca9ce430ec30dc8e40739194ee2b5860f71

nairbv committed:
`torch.pow` Add type promotion support and fix issue with __rpow__ (#37098)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/37098

### **Cherry-picked from another stack:**
Some code review already occurred here: https://github.com/pytorch/pytorch/pull/32582

### Summary:

Fixes: https://github.com/pytorch/pytorch/issues/32436

The issue caused incorrect handling of dtypes for scalar ** tensor.
e.g. before this change:
```
>>> 5.5 ** torch.ones(5, dtype=torch.int32)
tensor([5, 5, 5, 5, 5], dtype=torch.int32)
```
should return a float tensor.

Also fixes a number of incorrect cases:
 * tensors to negative powers were giving incorrect results (1 instead
    of 0 or error)
 * Behavior wasn't consistent between cuda/cpu
 * large_value ** 1 in some cases gave a result not equal
    to large_value because of truncation in conversion to double and back.

BC-breaking:

Previously incorrect behavior (in 1.4):
```
>>> a
tensor([1, 1, 1, 1, 1], dtype=torch.int32)
>>> a.pow_(.5)
tensor([1, 1, 1, 1, 1], dtype=torch.int32)
```

After this change:
`RuntimeError: result type Float can't be cast to the desired output type Int`

Test Plan: Imported from OSS

Differential Revision: D21686207

Pulled By: nairbv

fbshipit-source-id: e797e7b195d224fa46404f668bb714e312ea78ac

Commit:
improve accuracy of logsoftmax computation on cuda (#38945)

Summary:
Fixes https://github.com/pytorch/pytorch/issues/38839. Previously, if magnitude of input values was large, when computing `max+log(sum)` the `log(sum)` value was essentially ignored, now the result is computed as
`x-max-log(sum)` which has a better chance of preserving accuracy.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38945

Differential Revision: D21712483

Pulled By: ngimel

fbshipit-source-id: c1a3599ed981ba7a7fd130cbd7040a706b7eace0

Commit:
Updating submodules

Summary:
GitHub commits:

https://github.com/facebook/fbthrift/commit/3491869253e9acccf0bdea39eb8272eb85d96d32
https://github.com/pytorch/fbgemm/commit/56b6191c137fc01f4bf277566ef479b2bd88a04e

Test Plan: n/a

Reviewed By: 2d2d2d2d2d

fbshipit-source-id: a117b779500b040f5c1e087ed4ffb1587d745663

guyang3532 committed:
Use workspace to persist and restore images for Windows CI build and … (#38971)

Summary:
Inspired by malfet

> By the way, once we have build_artifacts property, can someone try if its faster to use it as mean of transferring images between build and test instead of using AWS (i.e. use artifacts instead of jenkins/pytorch/win-test-helpers/upload_image.py /download_image.py pair)

Use CircleCI to store intermediate binaries and make them available to be downloaded as artifacts instead of uploading to S3.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38971

Differential Revision: D21717080

Pulled By: seemethere

fbshipit-source-id: e3498b058778d02ae2f38daefbc7118a1a2cbe76

IvanKobzarev committed:
[Mobile GPU][Integration] Vulkan backend integration (#36491)

Summary:
This PR contains the initial version of Vulkan (GPU) Backend integration.
The primary target environment is Android, but the desktop build is also supported.

## CMake
Introducing three cmake options:
USE_VULKAN:
The main switch, if it is off, all other options do not affect.
USE_VULKAN_WRAPPER:
ON - Vulkan will be used loading it at runtime as "libvulkan.so" using libdl, every function call is wrapped in vulkan_wrapper.h.
OFF - linking with libvulkan.so directly
USE_VULKAN_SHADERC_RUNTIME:
ON - Shader compilation library will be linked, and shaders will be compiled runtime.
OFF - Shaders will be precompiled and shader compilation library is not included.

## Codegen
if `USE_VULKAN_SHADERC_RUNTIME` is ON:
Shaders precompilation () starts in cmake/VulkanCodegen.cmake, which calls `aten/src/ATen/native/vulkan/gen_glsl.py` or `aten/src/ATen/native/vulkan/gen_spv.py` to include shaders source or SPIR-V bytecode inside binary as uint32_t array in spv.h,spv.cpp.
if `USE_VULKAN_SHADERC_RUNTIME` is OFF:
The source of shaders is included as `glsl.h`,`glsl.cpp`.

All codegen results happen in the build directory.

## Build dependencies
cmake/Dependencies.cmake
If the target platform is Android - vulkan library, headers, Vulkan wrapper will be used from ANDROID_NDK.
Desktop build requires the VULKAN_SDK environment variable, and all vulkan dependencies will be used from it.
(Desktop build was tested only on Linux).

## Pytorch integration:
Adding 'Vulkan" as new Backend, DispatchKey, DeviceType.
We are using Strided layout without supporting strides at the moment, but we plan to support them in the future.
Using OpaqueTensorImpl where OpaqueHandle is copyable VulkanTensor,
more details in comments in `aten/src/ATen/native/vulkan/Vulkan.h`

Main code location: `aten/src/ATen/native/vulkan`
`aten/src/ATen/native/vulkan/VulkanAten.cpp` - connection link between ATen and Vulkan api (Vulkan.h) that converts at::Tensor to VulkanTensor.

`aten/src/ATen/native/Vulkan/Vulkan.h` - Vulkan API that contains VulkanTensor representation and functions to work with it. Plan to expose it for clients to be able to write their own Vulkan Ops.

`aten/src/ATen/native/vulkan/VulkanOps.cpp` - Vulkan Operations Implementations that uses Vulkan.h API

## GLSL shaders
Located in `aten/src/ATen/native/vulkan/glsl` as *.glsl files.
All shaders use Vulkan specialized constants for workgroup sizes with ids 1, 2, 3

## Supported operations
Code point:
conv2d no-groups
conv2d depthwise
addmm
upsample nearest 2d
clamp
hardtanh

## Testing
`aten/src/ATen/test/vulkan_test.cpp` - contains tests for
copy from CPU to Vulkan and back
all supported operations
Desktop builds supported, and testing can be done on a desktop that has Vulkan supported GPU or with installed software implementation of Vulkan, like https://github.com/google/swiftshader

## Vulkan execution
The initial implementation is trivial and waits every operator's execution.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/36491

Differential Revision: D21696709

Pulled By: IvanKobzarev

fbshipit-source-id: da3e5a770b1a1995e9465d7e81963e7de56217fa

linbinyu committed:
add eq.str op to lite interpreter (#38859)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38859

This error message indicates aten::eq expects different types

```
RUNNING 379 OP 76, aten::eq
terminate called after throwing an instance of 'c10::Error'
  what():  isInt() INTERNAL ASSERT FAILED at "buck-out/gen/68e83026/xplat/caffe2/aten_header#header-mode-symlink-tree-with-header-map,headers/ATen/core/ivalue.h":331, please report a bug to PyTorch.
```

It turns out that there are two aten::eq in lite interpreter (https://www.internalfb.com/intern/diffusion/FBS/browse/master/xplat/caffe2/torch/csrc/jit/runtime/register_prim_ops.cpp?lines=417)
aten::eq(int, int)
aten::eq(str, str)

This diff add overload name for str and it fixed the problem.

Test Plan: local test

Reviewed By: pengtxiafb

Differential Revision: D21681838

fbshipit-source-id: 1f17ecdadb9bc1c16915a24c60fa57a6fc273865

Commit:
Updates assertEqual to require atol and rtol, removes positional atol (#38872)

Summary:
This updates assertEqual and assertEqual-like functions to either require both or neither of atol and rtol be specified. This should improve clarity around handling precision in the test suite, and it allows us to remove the legacy positional atol argument from assertEqual. In addition, the "message" kwarg is replace with a kwarg-only "msg" argument whose name is consistent with unittest's assertEqual argument.

In the future we could make "msg" an optional third positional argument to be more consistent with unittest's assertEqual, but requiring it be specified should be clear, and we can easily update the signature to make "msg" an optional positional argument in the future, too.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38872

Differential Revision: D21717199

Pulled By: mruberry

fbshipit-source-id: 9feb856f94eee911b44f6c7140a1d07c1b026d3a

fnst-yule committed:
fix #38032 to allow specifying num_samples to RandomSampler when replacement is False

peterjc123 committed:
Expose VC_YEAR in Windows binary test jobs (#38957)

Summary:
To make it configurable  in https://github.com/pytorch/builder/pull/445.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38957

Differential Revision: D21721933

Pulled By: ezyang

fbshipit-source-id: 510b19e59bed4ff9d6c39173b4d5c5fc69290ed0

peterjc123 committed:
Disable some unsupported module for 32-bit build (#38950)

Summary:
Fixes https://github.com/pytorch/pytorch/issues/38322#issuecomment-632976523 and https://github.com/pytorch/pytorch/issues/38322#issuecomment-628698852.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38950

Differential Revision: D21721918

Pulled By: ezyang

fbshipit-source-id: 999788bb88d3e3c2c06f8dec4f0d6b3389095936

ssnl reviewed (changes requested):
Thanks. Before we can merge this, this needs (1) check that `num_samples <= dataset_size`, and (2) test.

zasdfgbnm committed:
[PATCH]  Migrate min from THC to ATen and remove _min (#38440)

Summary:
Related issue: https://github.com/pytorch/pytorch/issues/36900

Since I feel this PR is already large enough, I didn't migrate max in this PR. Legacy code is not cleaned up either. All these remaining work will be done in later PRs after this is merged.

Benchmark on an extreme case
```python
import torch
print(torch.__version__)

t = torch.randn(100000, 2, device='cuda')

warmup = torch.arange(100000000)
torch.cuda.synchronize()

%timeit t.min(dim=0); torch.cuda.synchronize()
```
Before: 4ms; After: 24.5us.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38440

Differential Revision: D21560691

Pulled By: ngimel

yns88 committed:
Update ShipIt sync

fbshipit-source-id: 945b4dfe99016e1788d0ec5429343e8c610f4e20

fnst-yule commented:
I'll fix that

diego-plan9 committed:
Use a temporary file during ReducerTest (#39004)

Summary:
Fixes https://github.com/pytorch/pytorch/issues/37558

Use a temporary file instead of `/dev/null` in `ReducerTest`, to prevent the chance of unintended deletion when running as root. It seemed that there were no strong side-effects (running time?) by fixing it at the test level, compared to other solutions that involved modifying the behaviour of `FileStore` (for example, adding an optional flag to avoid auto-deleting the file upon destruction).

Please note this is my first contribution - I have done my best to read the contributing guide and checked for duplicate PRs with no luck, but apologies in advance for any oversights and lack of familiarity with the procedures.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/39004

Differential Revision: D21721966

Pulled By: mrshenli

fbshipit-source-id: 76fb81600fa08a91c35d0eb9a5aab179f5371422

Commit:
move int8 fc operators and dependencies (#38935)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38935

move int8 ops

Test Plan: sandcastle

Reviewed By: jackm321, zrphercule

Differential Revision: D21704235

fbshipit-source-id: 7d5b570a5840ff21ffa6256604f892a084a30b31

ShawnZhong committed:
Remove reference of CUDA < 9.2 (#38977)

Summary:
Since CUDA < 9.2 is no longer supported (See https://github.com/pytorch/pytorch/pull/36848, https://github.com/pytorch/pytorch/pull/36846), this PR updates the required CUDA version in README.md to avoid confusion.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38977

Differential Revision: D21722965

Pulled By: ezyang

fbshipit-source-id: 626772f4303d023918dda34a620d95693174d97f

IvanKobzarev committed:
[vulkan] Fix Bazel build, add aten/native/vulkan/stub/*.cpp (#39018)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/39018

Differential Revision: D21724853

Pulled By: IvanKobzarev

fbshipit-source-id: 8d5bbc914b168da7d27c5447d625a9cfce61127f

xwang233 committed:
Fix max_pool2d for returning wrong shape with return_indices=True on cuda (#38992)

Summary:
Fix https://github.com/pytorch/pytorch/issues/38986

The current code only resizes pooling output but forget to resize indices as well.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38992

Differential Revision: D21718324

Pulled By: ngimel

fbshipit-source-id: 7cf937966d38ab2167be79979475c4e0cacbf82c

IvanKobzarev committed:
[vulkan] Fix python overrides tests for is_vulkan_available (#39016)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/39016

Differential Revision: D21724619

Pulled By: IvanKobzarev

fbshipit-source-id: d7a6c8b944a55bc4f2cce957eeac08c5801667a0

kshitij12345 committed:
Port TH cum{sum,prod}_cuda to ATen (#36458)

Summary:
References: https://github.com/pytorch/pytorch/issues/24521 #24522 https://github.com/pytorch/pytorch/issues/24547 #24548 https://github.com/pytorch/pytorch/issues/24507

Depends on https://github.com/pytorch/pytorch/issues/36308

Changes related to this PR are only in file :
aten/src/ATen/Declarations.cwrap
aten/src/ATen/native/cuda/ReduceOpsKernel.cu
aten/src/ATen/native/native_functions.yaml
aten/src/THC/generic/THCTensorMathScan.cu
aten/src/THC/generic/THCTensorMathScan.h

Please Review VitalyFedyunin

Thanks.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/36458

Differential Revision: D21718384

Pulled By: ngimel

fbshipit-source-id: 5af15164050c77be164397abd659a48c9ded2b29

xwang233 committed:
Fix max_pool2d nchw backward bug (#38953)

Summary:
Fix https://github.com/pytorch/pytorch/issues/38764

The current problem is that, `top_diff` and `top_mask` pointers are shifted "accumulatively" with for-n and for-c loops. This may cause overflow and illegal memory access when the loop counts are greater than one, that is n > 65535 or c > 65535 (the case in https://github.com/pytorch/pytorch/issues/38764). Since neither of n > 65535 or c > 65535 is common, it has not been seen before. The simple fix would be using new pointer variables for the n & c offset instead of directly modifying `top_diff` or `top_mask`.

However, I think the current nchw max_pool2d GPU impl still has plenty of room for performance improvement. We can check that in a later PR if needed.

Slightly clean up the indentation. Also add tests to use CPU impl as a reference check.

cc skrah
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38953

Differential Revision: D21721930

Pulled By: ezyang

fbshipit-source-id: fef7d911d814f8ed9fd67c60cabe5d52f8fd3d57

rohan-varma committed:
Revert D21681838: add eq.str op to lite interpreter

Test Plan: revert-hammer

Differential Revision:
D21681838

Original commit changeset: 1f17ecdadb9b

fbshipit-source-id: bac620957d1a68057cd53f91f6a837d2b64f0e5e

Commit:
[c2] cuda impl for WeightScale op (#38712)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38712

as title

Test Plan: buck test;

Reviewed By: ustctf

Differential Revision: D21586705

fbshipit-source-id: 12cd34f04f074ee12b77304055f3ba6068cf38fb

ShawnZhong committed:
Fix histc with empty tensor error (#38987)

Summary:
Fix https://github.com/pytorch/pytorch/issues/38979

The error in mentioned https://github.com/pytorch/pytorch/issues/38979 is a [`cudaErrorInvalidConfiguration` error](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html#group__CUDART__TYPES_1g3f51e3575c2178246db0a94a430e0038):
> This indicates that a kernel launch is requesting resources that can never be satisfied by the current device. Requesting more shared memory per block than the device supports will trigger this error, as will requesting too many threads or blocks. See cudaDeviceProp for more device limitations.

This is because we are trying to launch a kernel with block size 0.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38987

Differential Revision: D21722993

Pulled By: ezyang

fbshipit-source-id: 2c283e0a9f542b4acb96e895a43b991ccac808fe

mruberry committed:
Updates torchvision version (#38848)

Summary:
In PyTorch 1.6 integer division using torch.div will throw a runtime error. When PyTorch Master adopts this behavior some of our ONNX tests would break if we continued to import torchvision v0.5, since v0.5 uses torch.div to perform integer division. fmassa and I recently updated Torchvision to use torch.floor_divide for integer division (at least on paths covered by the PyTorch OSS CI tests), and this PR updates our torchvision test version to include those changes. This will prevent the PyTorch OSS CI from breaking when PyTorch Master adopts the 1.6 integer division behavior.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38848

Differential Revision: D21679988

Pulled By: mruberry

fbshipit-source-id: 1333f6254c295909cf05b6f3e352e4a0c336e5af

jeffdaily committed:
Add ROCm-specific half_support_literal for JIT. (#38899)

Summary:
CC ezyang xw285cornell sunway513 lcskrishna
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38899

Differential Revision: D21721855

Pulled By: ezyang

fbshipit-source-id: 3739c462f04cee40ff979f44387ef66b971f5303

malfet committed:
Windows build updates (#39035)

Summary:
Small follow-up updates https://github.com/pytorch/pytorch/pull/38971
- Remove extra whitespace
- Delete unused script files
- Modify `vs_install.ps1` to use correct workspace path
Pull Request resolved: https://github.com/pytorch/pytorch/pull/39035

Differential Revision: D21731129

Pulled By: malfet

fbshipit-source-id: e04253e82f8753423b4634d1928f2d0fcf20ebbb

mattip committed:
DOC: fix import error (#38921)

Summary:
Fixes errors when importing the module. The import is used by sphinx in documentation builds.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38921

Differential Revision: D21722144

Pulled By: ezyang

fbshipit-source-id: 5f31d4750325f1753de93754a009006cbc13655e

kshitij12345 committed:
[docs] fix formula `torch.logcumsumexp` (#38952)

Summary:
Reference : https://github.com/pytorch/pytorch/pull/36308#issuecomment-632282641

After fix:

![Screenshot from 2020-05-23 15-35-09](https://user-images.githubusercontent.com/19503980/82727956-4bcabb80-9d0b-11ea-85a8-81b35012abbc.png)
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38952

Differential Revision: D21722196

Pulled By: ezyang

fbshipit-source-id: 62b08c14e0ce9603133841940627df40d7b1e861

Commit:
Add index number to THArgCheck error message (#38978)

Summary:
- Resolving the feature introduced in https://github.com/pytorch/pytorch/issues/38652
- Since the iteration will be terminated once the error occurred, perhaps we can only give the current index which caused the error.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38978

Differential Revision: D21722426

Pulled By: ezyang

fbshipit-source-id: edfc3f7a320584ba22d790f2b79c3726e99aae2a

mattip committed:
make onnx expect tests resiliant to producer_version changes (#39002)

Summary:
closes gh-32561 closes gh-38545. As part of the fallout from gh-36797, this PR
- replaces the producer_version: "1.6" in onnx expect tests with `producer_version: "XXX"
- adapts `testing/_internal/common_utils.py` with a regex to change the onnx producer_version so tests still pass

The consistency of the torch version and the onnx `producer_version` is tested in gh-36797, so there is no reason to test it again in the expect tests.

xref gh-38629 which documented how to run the onnx tests and at the same time refactored the Community documentation.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/39002

Differential Revision: D21723062

Pulled By: ezyang

fbshipit-source-id: 1bd6a8ed37d5383e69d017226dc09c0645a69aff

chengjinfang1216 committed:
Fix the issue that Bad interaction between no_grad and numpy conversi… (#38906)

Summary:
…on(https://github.com/pytorch/pytorch/issues/37000)
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38906

Differential Revision: D21722033

Pulled By: albanD

fbshipit-source-id: f22aec8106e4546e828aba15be606e9d9f3eeffa

IvanKobzarev committed:
[vulkan] TensorConversions remove explicit vulkan ifs (#39019)

Summary:
As a follow up for https://github.com/pytorch/pytorch/pull/36491 and last comments on it.

Vulkan uses Strided Layout (at the moment strides are not supported, but in plan)
empty_strided just forwards to empty_vulkan, ignoring strides params.

Removing explicit ifs in TensorConversions that were added before decision to use Strided layout and have not been cleaned after that :(
Pull Request resolved: https://github.com/pytorch/pytorch/pull/39019

Differential Revision: D21726480

Pulled By: IvanKobzarev

fbshipit-source-id: d465456df248a118bfef441c85280aa0025860cd

andyljones committed:
Removes pickle deprecation warning (#39003)

Summary:
As per [this issue](https://github.com/pytorch/pytorch/issues/38597), this is a one-line PR to remove the pickle deprecation warning.

cc stsievert driazati
Pull Request resolved: https://github.com/pytorch/pytorch/pull/39003

Differential Revision: D21723048

Pulled By: ezyang

fbshipit-source-id: f6cc8f28b8140edd7b46d1f26f2d99819beb933e

peterjc123 committed:
Migrate Windows build jobs to VS 2019 for CUDA >= 10.1 (#38959)

Summary:
This PR relies on https://github.com/pytorch/pytorch/pull/38957 and https://github.com/pytorch/builder/pull/445.
Tested with pytorch/pytorch#38949 and pytorch/pytorch#38956.
Will need a rebase after the dependent commits go in
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38959

Differential Revision: D21732423

Pulled By: malfet

fbshipit-source-id: 50837a026a575bb3d547526e299db7bcfd7637a8

Commit:
restore proper cuda assert behavior with DNDEBUG (#38943)

Summary:
Per title. https://github.com/pytorch/pytorch/issues/32719 essentially disabled asserts in cuda kernels in release build. Asserts in cuda kernels are typically used to prevent invalid reads/writes, so without asserts invalid read/writes are silent errors in most cases (sometimes they would still cause "illegal memory access" errors, but because of caching allocator this usually won't happen).
We don't need 2 macros, CUDA_ALWAYS_ASSERT and CUDA_KERNEL_ASSERT because all current asserts in cuda kernels are important to prevent illegal memory accesses, and they should never be disabled.
This PR removes macro CUDA_ALWAYS_ASSERT and instead makes CUDA_KERNEL_ASSERT (that is commonly used in the kernels) an asserttion both in release and debug builds.
Fixes https://github.com/pytorch/pytorch/issues/38771
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38943

Differential Revision: D21723767

Pulled By: ngimel

fbshipit-source-id: d88d8aa1b047b476d5340e69311e65aff4da5074

rohan-varma committed:
Revert D21717199: [pytorch][PR] Updates assertEqual to require atol and rtol, removes positional atol

Test Plan: revert-hammer

Differential Revision:
D21717199

Original commit changeset: 9feb856f94ee

fbshipit-source-id: bfde9c39a5ce99f0ca6183a7dde703c65b7c8259

Krovatkin committed:
run te only for profiling executor (#38591)

Summary:
* Disable the mode where PE can still run the old fuser.
* Clean up
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38591

Differential Revision: D21643664

Pulled By: Krovatkin

fbshipit-source-id: 6753ed6bdc544698a1340e59a624608ff3abf7f9

nickgg committed:
[TensorExpr] Fix elimination of For loops with empty bodies (#38883)

Summary:
We do try to eliminate empty For loops, but missed a case where the body Block exists but is empty. In that case we can eliminate the loop as well.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38883

Differential Revision: D21723680

Pulled By: nickgg

fbshipit-source-id: 49610b0524af5b9ec30ef3b4cc0c8461838259c3

rohan-varma committed:
Remove unneeded const from process group agent header (#38804)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38804

This is only needed in the process group agent implementation, and
removing it from the header file prevents other translation units that include
it from having this constant.
ghstack-source-id: 104666599

Test Plan: CI

Differential Revision: D21668514

fbshipit-source-id: 1c39cc98dea99518134c66dca3ca5b124a43de1b

houseroad committed:
Automatic update of fbcode/onnx to 20b3e10e6c3a9cdab90d2bb864d1c36d3e3651cd (#38203)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38203

Previous import was 79a7e0df7e86e0f32e7a05f563b24a566540c18b

Included changes:
- **[20b3e10e](https://github.com/onnx/onnx/commit/20b3e10e)**: Add 'ignore_index' input in the spec for SoftmaxCrossEntropyLoss and NLLLoss. (#2680) <M. Zeeshan Siddiqui>
- **[eae3eb8c](https://github.com/onnx/onnx/commit/eae3eb8c)**: Use cmake GNUInstallDirs (#2661) <Gustavo Alvarez>
- **[106821e9](https://github.com/onnx/onnx/commit/106821e9)**: Update sequence test case so input is not scalar and splits are specified (#2675) <Scott McKay>
- **[e094e101](https://github.com/onnx/onnx/commit/e094e101)**: Remove unnecessary copies and std::move (#2684) <Changming Sun>
- **[71145275](https://github.com/onnx/onnx/commit/71145275)**: Update Batchnorm test (#2674) <Lara Haidar>
- **[da13be2d](https://github.com/onnx/onnx/commit/da13be2d)**: Rename OPTIONAL to OPTIONAL_VALUE (#2682) <Changming Sun>
- **[2987fa06](https://github.com/onnx/onnx/commit/2987fa06)**: Adding CI for ONNX Debug mode (Linux, OSX) (#2651) <Vinitra Swamy>
- **[46fe392d](https://github.com/onnx/onnx/commit/46fe392d)**: Update Pow input types in Opset 12 (#2666) <Lara Haidar>
- **[ac1caf3b](https://github.com/onnx/onnx/commit/ac1caf3b)**: Change type of label tensor to int32/int64 in SoftmaxCrossEntropyLoss spec. (#2667) <M. Zeeshan Siddiqui>
- **[c2fefcbf](https://github.com/onnx/onnx/commit/c2fefcbf)**: [Training] SG with Momentum Optimizer (#1959) <Wei-Sheng Chin>
- **[8d15705e](https://github.com/onnx/onnx/commit/8d15705e)**: [Training] Add Adagrad optimizer operator (#1955) <Wei-Sheng Chin>
- **[94b01cdd](https://github.com/onnx/onnx/commit/94b01cdd)**: Suppress a warning in unsqueeze (#2637) <Hong Xu>
- **[0582d526](https://github.com/onnx/onnx/commit/0582d526)**: Fix Greater/LessOrEqual function definition (#2645) <Takeshi Watanabe>
- **[b852d819](https://github.com/onnx/onnx/commit/b852d819)**: Increment version number to 1.7.0 (#2639) <Chin Huang>
- **[ff4bb553](https://github.com/onnx/onnx/commit/ff4bb553)**: Regenerate Min test data (#2644) <Takeshi Watanabe>

Test Plan: ci

Reviewed By: hl475

Differential Revision: D21493165

fbshipit-source-id: 6863b289bfbf4235e36f0e2456ce44c776aaf164

supriyar committed:
[quant] Onnx export of quantized models with new API (#38736)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38736

qconv2d and qlinear APIs were changed recently so updating the scale code accordingly

Test Plan:
python test/onnx/test_pytorch_onnx_caffe2_quantized.py

Imported from OSS

Differential Revision: D21647724

fbshipit-source-id: 45d4b358ffb84f1e73da8ba3f702d5043bdb16d2

valgur committed:
Fix torch.hub.hub_dir inconsistencies (#38969)

Summary:
Fixes https://github.com/pytorch/pytorch/issues/38401

* `torch.hub.load_state_dict_from_url()` now also downloads to `$TORCH_HOME/hub/checkpoints` instead of `$TORCH_HOME/checkpoints` like `torch.hub.load()` and others.
* Make `hub_dir` private, add and use `get_dir()` instead.

Also updated docs. Did not see a need for additional unit tests.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38969

Differential Revision: D21725880

Pulled By: ailzhang

fbshipit-source-id: 58cc6b32ddbda91e58c1c1433cc3916223556ea1

Commit:
Revert D21493165: Automatic update of fbcode/onnx to 20b3e10e6c3a9cdab90d2bb864d1c36d3e3651cd

Test Plan: revert-hammer

Differential Revision:
D21493165

Original commit changeset: 6863b289bfbf

fbshipit-source-id: 47b530c8ffceb3673a86b6cf0c064fe6af0eb72d

Commit:
[JIT] Allow @torch.jit.unused to be used on TS classes (#38522)

Summary:
**Summary**
This commit enables the use of `torch.jit.unused` on methods of TorchScript classes.
This attribute is honoured by replacing the body of any method
marked as unused in the parsed AST for the class with `raise Exception(...)`.

**Test Plan**
This commit adds a unit test `TestClassType.test_unused_method` that
tests this feature.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38522

Differential Revision: D21733818

Pulled By: SplitInfinity

fbshipit-source-id: 771872359dad70fac4aae83b6b5f17abb6329890

Krovatkin committed:
move num_profiled_runs to common_utils (#38687)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/38687

Differential Revision: D21634080

Pulled By: Krovatkin

fbshipit-source-id: 55513124caf3885e475ffecd9d9f3dbc4729a573

Commit:
Updates assertEqual to require atol and rtol, removes positional atol (#38872)

Summary:
This updates assertEqual and assertEqual-like functions to either require both or neither of atol and rtol be specified. This should improve clarity around handling precision in the test suite, and it allows us to remove the legacy positional atol argument from assertEqual. In addition, the "message" kwarg is replace with a kwarg-only "msg" argument whose name is consistent with unittest's assertEqual argument.

In the future we could make "msg" an optional third positional argument to be more consistent with unittest's assertEqual, but requiring it be specified should be clear, and we can easily update the signature to make "msg" an optional positional argument in the future, too.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38872

Differential Revision: D21740237

Pulled By: mruberry

fbshipit-source-id: acbc027aa1d7877a49664d94db9a5fff91a07042

gchanan committed:
Improve cpu/Loops.h arity asserts. (#38809)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38809

This splits the asserts into separate input/output asserts and makes the numbers precise, instead of ranges.

This is an ongoing effort to improve the Loops assertion and to integrate dynamic cast checking into CPU loops.

Test Plan: Imported from OSS

Differential Revision: D21670263

Pulled By: gchanan

fbshipit-source-id: b1868db5255a69158045b759dc9171690a2dcd01

gchanan committed:
Support void return type in TensorIteratorDynamicCasting checks. (#38815)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38815

Some CPU kernels have void return types and the currently implementation segfaults on these cases.

Test Plan: Imported from OSS

Differential Revision: D21670717

Pulled By: gchanan

fbshipit-source-id: bc17b8330195601ca231a985ee44319447ba6cf0

gchanan committed:
Improve error checking of CUDALoops. (#38810)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38810

Same change as was applied to CPU loops -- separate out checking of the inputs and outputs.

Test Plan: Imported from OSS

Differential Revision: D21670339

Pulled By: gchanan

fbshipit-source-id: 42f208538dce1a5598d14948d8d02a1c91ba152a

gchanan committed:
Move needs_dynamic_casting to a non-CUDA specific file. (#38813)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38813

We are going to apply this check to CPU (with some changes), so just moving this in preparation.

The code is just cut-pasted here, no behavioral change.

Test Plan: Imported from OSS

Differential Revision: D21670554

Pulled By: gchanan

fbshipit-source-id: c7e07f67bb4c6524fde12237e35892e42557103e

albanD committed:
Update to use the stable windows image instead of the temporary one (#39066)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/39066

Differential Revision: D21742331

Pulled By: albanD

fbshipit-source-id: e4184b38eb69a289910e79808ebe9b2510dc6b06

neginraoof committed:
[ONNX] Enable models tests (#38791)

Summary:
PR to enable model tests which are fixed.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38791

Reviewed By: hl475

Differential Revision: D21732498

Pulled By: houseroad

fbshipit-source-id: f417f9d4124ef5a663dc666d5c2ed6ba013b26a4

peterjc123 committed:
Fix Windows binary jobs after migrating to the new circleci image (#39057)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/39057

Differential Revision: D21742971

Pulled By: albanD

fbshipit-source-id: a25ab8b01a9b7c1e2d14fe38227f85a5b8f0db83

jeffdaily committed:
[ROCm] HIP version guard for occupancy API compatibility (#38551)

Summary:
CC ezyang xw285cornell

HIP from ROCm 3.5 renames `hipOccupancyMaxActiveBlocksPerMultiprocessor` to `hipModuleOccupancyMaxActiveBlocksPerMultiprocessor`.  In addition, the API parameter types now match CUDA.  Add these changes in a backwards-compatible manner.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38551

Differential Revision: D21721832

Pulled By: ezyang

fbshipit-source-id: 6fc971845e363d7495d8be9550e76d0f082c3062

osalpekar committed:
[TensorPipe Agent] Improve Response Error Message on Agent Shutdown (#38818)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38818

Standardizes the error message when a response is attempted after the
agent has shut down.
ghstack-source-id: 104673115

Test Plan: Sandcastle - no functionality change, just error message

Differential Revision: D21670706

fbshipit-source-id: d26fcd7c76758c62d432d9c4e6ef2e3af7cbedff

osalpekar committed:
[TensorPipe Agent] Message on Agent Shutdown (#38819)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38819

Logs a message when the agent is shutting down like the other RPC
Agents.
ghstack-source-id: 104673386

Test Plan: Sandcastle

Differential Revision: D21671061

fbshipit-source-id: a44f0e4976e3acc898645a2baf6f41f45a697166

KsenijaS committed:
Enable Constant Folding Tests (#38751)

Summary:
Enable tests for constant folding since constant folding is enabled for opset 12
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38751

Differential Revision: D21728013

Pulled By: ezyang

fbshipit-source-id: e0ed9ad62d8b781eacfdf894e8c9609fe7e778bd

ssnl committed:
Fix broken reference in sync bn doc (#38890)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/38890

Differential Revision: D21722162

Pulled By: ezyang

fbshipit-source-id: a7d18239917b2886fe8c1c0aaf42fc8491c8e10c

Commit:
Use ninja build as default for HIPExtensions (#38939)

Summary:
This PR adds the following changes:
1. It sets the default extension build to use ninja
2. Adds HIPCC flags to the host code compile string for ninja builds. This is needed when host code makes HIP API calls

cc: ezyang jeffdaily
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38939

Differential Revision: D21721905

Pulled By: ezyang

fbshipit-source-id: 75206838315a79850ecf86a78391a31ba5ee97cb

Commit:
Change len(DataLoader) for IterableDataset (#38925)

Summary:
Fix https://github.com/pytorch/pytorch/issues/36176

One-liner change to ensure that ```len(loader) == (len(dataset) // batch_size)``` for IterableDataset.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38925

Differential Revision: D21731587

Pulled By: ezyang

fbshipit-source-id: 59a086165a004c0c1c8a1ee0776b1444bd26de23

peterbell10 committed:
Fix attribute warning on gcc (#38988)

Summary:
When building, my log was being spammed with:
```
warning: attribute "__visibility__" does not apply here
```

Which, at least on gcc 7.4 isn't covered by silencing `-Wattribute`. The warning suggests `enum`s don't need to be exported on linux, so I just `ifdef` it out instead.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38988

Differential Revision: D21722032

Pulled By: ezyang

fbshipit-source-id: ed4cfebc187dceaa9e748d85f756611fd7eda4b4

nairbv committed:
Simplify precision-specification in tests. (#37181)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/37181

Now that assertEquals considers dtypes in determining tolerance, most
tests don't need explicitly set precision.

Those that do are a few half precision tests on cuda. In this PR, those
are broken out to be handled explicitly, though we may also want to
consider further loosening the tolerance on half-precision.

Test Plan: Imported from OSS

Differential Revision: D21728402

Pulled By: nairbv

fbshipit-source-id: 85f3daf63f1bdbb5101e8dea8c125f13448ca228

Commit:
change the int8 test to use unquantized bias (fp32)

Summary: change the test default to test the version we care about

Test Plan: ran the test

Reviewed By: amylittleyang

Differential Revision: D21725194

fbshipit-source-id: 243fcdf1dd5784768f6ceb2b46f9f1c9e64341eb

nikitaved committed:
remove serial_exec from scatter/gather kernel (#36181)

Summary:
Since the indexed dimension in `scatter/gather` is traversed inside the kernel, all the memory conflicts of writing to the same memory between the threads are actually mutually disjoint.
See [this comment](https://github.com/pytorch/pytorch/issues/33389#issuecomment-590017938) for a graphical explanation. More formal description:
Suppose we deal with 3D tensors and `dim=0`, hence the `scatter_add` operations are
```
self[index[i][j][k]][j][k] += src[i][j][k],
...
self[index[i'][j'][k']][j'][k'] += src[i'][j'][k'],
...
```
Clearly, write/read to the same memory happens if and and only if:
```
index[i][j][k] = index[i'][j'][k'],
j = j',
k = k'.
```
Since the reduction over `dim=0` happens inside the kernel, threads `i` and `i'` partition `dim=1,2`. It means that threads `i` and `i'` receive indices
```
I = {(*, i, k) sent to the thread i},
I' = {(*, i', k') sent to the thread i'},
I intersection with I' = the empty set.
```

This happens:
```
index[i][j][k] = index[i'][j'][k'],
j = j',
k = k',
```
if and only if there exists some thread k which receives indices K and
`(*,j,k),(*,j',k') in K`.

Therefore it is possible to make `scatter_add` parallel and remove `serial_exec` from the `scatter_gather_base_kernel`.

CC v0dro
Pull Request resolved: https://github.com/pytorch/pytorch/pull/36181

Differential Revision: D21716167

Pulled By: ngimel

fbshipit-source-id: 49aee2de43779a1f0b359c22c8589c0702ee68a2

chengjinfang1216 committed:
Fix the issue that PyTorch doesn't construct bool tensors from non-bo… (#38392)

Summary:
…ol values correctly(https://github.com/pytorch/pytorch/issues/37398)

Signed-off-by: chengjinfang <chengjf@cn.fujitsu.com>
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38392

Differential Revision: D21737009

Pulled By: mruberry

fbshipit-source-id: c77d8c940af95f5011fe008b48ea0d16c3f501d1

malfet committed:
[Codemod][GleanFbcode] Remove dead includes in caffe2/test (#39023)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/39023

Reviewed By: orionr

Differential Revision: D21702529

fbshipit-source-id: 6945bba95609102409850b105a8a091e33b8acc9

kimishpatel committed:
Add JIT fusion pass to fuse quantized add and relu. (#38897)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38897

Quantized ops support add_relu. This pass enables finding quantized add + relu
pattern and fuse them to add_relu.

Test Plan: buck run caffe2/test:quantization -- test_quantization.TestFusionPasses

Reviewed By: jerryzh168

Differential Revision: D21690909

fbshipit-source-id: 607cf72dde535df15eb7638841543ab2156af464

kimishpatel committed:
Expose qnnpack's maxpool when going through aten::max_pool2d (#38896)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38896

Current way of exposing qnnpack's maxpool2d only works if max_pool2d op is
quantized::max_pool2d. This diff moves the function about to expose it via
aten::max_pool2d when dispatch key is QuantizedCPU.

Test Plan: Quantized tests.

Reviewed By: supriyar

Differential Revision: D21690913

fbshipit-source-id: 75fb77329b915e3a3c3aac4d76359482976ca783

Commit:
Throws runtime error on attempted addcdiv integer division (#38762)

Summary:
1.6 Deprecation Note:

In 1.6 attempting to perform integer division using addcdiv will throw a RuntimeError, and in 1.7 the behavior will change so that addcdiv always performs a true division of its tensor1 and tensor2 inputs. See the warning in torch.addcdiv's documentation for more information.

PR Summary:

This PR updates the warning that appears when addcdiv performs integer division to throw a RuntimeError. This is intended to prevent silent errors when torch.addcdiv's behavior is changed to always perform true division in 1.7. The documentation is updated (slightly) to reflect this, as our the addcdiv tests in test_torch and test_type_promotion.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38762

Differential Revision: D21657585

Pulled By: mruberry

fbshipit-source-id: c514b44409706f2bcfeca4473424b30cc48aafbc

ChunliF committed:
[disagg_acc] In batch broadcast (#38700)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/38700

Reviewed By: yinghai

Differential Revision: D21634147

fbshipit-source-id: 7bd1912654e2433cfb580b5f7a9fb86570a55cab

ailzhang committed:
Replace use of np.allclose in tests. (#34287)

Summary:
fixes https://github.com/pytorch/pytorch/issues/34096
Pull Request resolved: https://github.com/pytorch/pytorch/pull/34287

Differential Revision: D21735525

Pulled By: ailzhang

fbshipit-source-id: 611da17cfc5a3fee77d482abccf8f9854f504263

durumu committed:
Refactor backward compatibility tests to use override_qengines decorator (#38838)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/38838

Test Plan: Imported from OSS

Differential Revision: D21676032

Pulled By: durumu

fbshipit-source-id: 5cbe56e0d72d322f540bccffb60bcdbb15385ee8

Commit:
[JIT][to-backend] Fix compilation unit and name mangling of generated module (#38679)

Summary:
**Summary**
This commit gets rid of the separate compilation unit that is currently
being created for every backend-specific module generated by
`jit::backend::generateToBackendFn` and mangles the name properly to
allow multiple backend-specific modules to coexist in the same
compilation unit.

**Test Plan**
`python test/test_jit.py TestBackends`

**Fixes**
This pull request fixes part of https://github.com/pytorch/pytorch/issues/37841.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38679

Differential Revision: D21744620

Pulled By: SplitInfinity

fbshipit-source-id: ac85b8ce0d179c057991e9299fd53a4e13ba02a9

malfet committed:
Test PyTorch using python-3.8 + GCC-9 on Bionic (#39030)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/39030

Differential Revision: D21752017

Pulled By: malfet

fbshipit-source-id: 56c841636349e24c9ebef8dac18c283de3664fa5

Commit:
Correct Javadoc link to master (#39038)

Summary:
Correct Javadoc link to match the 1.4 version: https://github.com/pytorch/pytorch/blob/release/1.4/docs/source/index.rst
Pull Request resolved: https://github.com/pytorch/pytorch/pull/39038

Differential Revision: D21747969

Pulled By: jlin27

fbshipit-source-id: 941b61204e9be53e15a6351eff6f4935e6a16d24

jeffdaily committed:
update circleci scripts for rocm ubuntu bionic support (#39097)

Summary:
CC ezyang xw285cornell sunway513
Pull Request resolved: https://github.com/pytorch/pytorch/pull/39097

Differential Revision: D21753340

Pulled By: ezyang

fbshipit-source-id: 9cd84e9c47c08702d7b67d071dc88c345b9db85c

nickgg committed:
[TensorExpr] Fix a bug in Rfactor when there are multiple reductions (#38733)

Summary:
In `LoopNest::rfactor` we assume that there is only a single reduction below the insertion point, and when replacing the reduction we recursively replace all reductions below that point. This is not a safe assumption, as a number of transformations can introduce additional ReduceOps - most directly a `splitWithTail` on the innermost reduce axis.

This PR fixes that bug, and adds some unit tests covering the case.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38733

Differential Revision: D21723634

Pulled By: nickgg

fbshipit-source-id: 3ed6ffcdc2c15aef7504f9b2b91e8d827e0b5d88

Commit:
Revert D21728402: Simplify precision-specification in tests.

Test Plan: revert-hammer

Differential Revision:
D21728402

Original commit changeset: 85f3daf63f1b

fbshipit-source-id: 4e2a36aca15cd8d842985173395b4e1cac7135d8

rohan-varma committed:
Revert D21752017: [pytorch][PR] Test PyTorch using python-3.8 + GCC-9 on Bionic

Test Plan: revert-hammer

Differential Revision:
D21752017

Original commit changeset: 56c841636349

fbshipit-source-id: adf08e03ba9610050fc5440ef453789f805fdc6b

houseroad committed:
Automatic update of fbcode/onnx to eae3eb8c61cf5ad27cc9a416dbdc5274982385a6 (#39089)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/39089

Previous import was 79a7e0df7e86e0f32e7a05f563b24a566540c18b

Included changes:
- **[eae3eb8c](https://github.com/onnx/onnx/commit/eae3eb8c)**: Use cmake GNUInstallDirs (#2661) <Gustavo Alvarez>
- **[106821e9](https://github.com/onnx/onnx/commit/106821e9)**: Update sequence test case so input is not scalar and splits are specified (#2675) <Scott McKay>
- **[e094e101](https://github.com/onnx/onnx/commit/e094e101)**: Remove unnecessary copies and std::move (#2684) <Changming Sun>
- **[71145275](https://github.com/onnx/onnx/commit/71145275)**: Update Batchnorm test (#2674) <Lara Haidar>
- **[da13be2d](https://github.com/onnx/onnx/commit/da13be2d)**: Rename OPTIONAL to OPTIONAL_VALUE (#2682) <Changming Sun>
- **[2987fa06](https://github.com/onnx/onnx/commit/2987fa06)**: Adding CI for ONNX Debug mode (Linux, OSX) (#2651) <Vinitra Swamy>
- **[46fe392d](https://github.com/onnx/onnx/commit/46fe392d)**: Update Pow input types in Opset 12 (#2666) <Lara Haidar>
- **[ac1caf3b](https://github.com/onnx/onnx/commit/ac1caf3b)**: Change type of label tensor to int32/int64 in SoftmaxCrossEntropyLoss spec. (#2667) <M. Zeeshan Siddiqui>
- **[c2fefcbf](https://github.com/onnx/onnx/commit/c2fefcbf)**: [Training] SG with Momentum Optimizer (#1959) <Wei-Sheng Chin>
- **[8d15705e](https://github.com/onnx/onnx/commit/8d15705e)**: [Training] Add Adagrad optimizer operator (#1955) <Wei-Sheng Chin>
- **[94b01cdd](https://github.com/onnx/onnx/commit/94b01cdd)**: Suppress a warning in unsqueeze (#2637) <Hong Xu>
- **[0582d526](https://github.com/onnx/onnx/commit/0582d526)**: Fix Greater/LessOrEqual function definition (#2645) <Takeshi Watanabe>
- **[b852d819](https://github.com/onnx/onnx/commit/b852d819)**: Increment version number to 1.7.0 (#2639) <Chin Huang>
- **[ff4bb553](https://github.com/onnx/onnx/commit/ff4bb553)**: Regenerate Min test data (#2644) <Takeshi Watanabe>

Test Plan: ci

Reviewed By: hl475

Differential Revision: D21750299

fbshipit-source-id: c33ec1b1e0dc65d0187e78db96d749f9037aae9c

cloudhan committed:
Implement logaddexp (#38384)

Summary:
Resolve https://github.com/pytorch/pytorch/issues/38377
Related https://github.com/pytorch/pytorch/issues/38349

This op should be disambiguated with `logsumexp` which do a reduction on a tensor over a specific axis.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38384

Differential Revision: D21737336

Pulled By: mruberry

fbshipit-source-id: 7864d04ca304c0fb2937bb083583e3e3d6ef205d

Commit:
[Caffe2] Change shape_hints format (#39100)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/39100

The old shape_hints format has a few cons:
- ',' is used to separate <model_id>:<shape_hints> pairs, as well as delimiter for dims in the <shape_hints>, which is an obvious bug
- it cannot handle the case of having ':' in tensor names

The new shape_hints format uses '::' to delimit <model_id> and <shape_hints>, ';' to delimit <model_id>::<shape_hints> pairs. Inside <shape_hints>, '|' is used to separate <tensor>,<shape> pairs, and ',' is used to delimit <tensor> and <shape>, as well as the dimensions inside <shape>.

Test Plan:
```
buck test //caffe2/caffe2/fb/opt:shape_info_utils_test
```

AI/AF canary:
https://www.internalfb.com/intern/ads/canary/426980448937212687
https://www.internalfb.com/intern/ads/canary/426980529105312403

Reviewed By: yinghai

Differential Revision: D21656832

fbshipit-source-id: 9dec4b5586d093ddb814c3f15041a57d45a3de76

rohan-varma committed:
Infinite timeout for operations against ProcessGroup for RPC (#38577)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/38577

We don't want to limit a timeout to 30 min since there could be no
operations within that time frame. Bump to 2^31 - 1 (int32 max)
ghstack-source-id: 104743727

Test Plan: CI

Differential Revision: D21602425

fbshipit-source-id: ab002262f01664b538761202b3bd7584fcee3c6b

malfet committed:
Get rid of multiple inheritence in test_torch (#39110)

Summary:
`_TestTorchMixin` is base class which is instantiated across multiple types.
It was inherited from `object` in order to hide it from unittest test discovery mechanism.
But this approach makes it almost impossible to use static code analyzer on the class.
This PR implements alternative approach by hiding base class into inner class, per https://stackoverflow.com/a/25695512

Change imported class access path in `test_cuda.py`
Pull Request resolved: https://github.com/pytorch/pytorch/pull/39110

Test Plan:
run `test_torch.py --discover-tests` and `test_cuda.py --discover-tests` before and after change:
```
$ python test_torch.py --discover-tests|md5sum
2ca437bb5d65700763ce04cdacf6de3e  -
$ python test_cuda.py --discover-tests|md5sum
b17df916fb0eeb6f0dd7222d7dae392c  -
```

Differential Revision: D21759265

Pulled By: malfet

fbshipit-source-id: b01b06111469e551f7b78387449975e5248f6b9e

fnst-yule commented:
I made a low-level mistake on git.
Sorry for bothering all of you.. :(
